src/main/clm_driver.F90

```diff
@@ -395,6 +399,10 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
        end if
        call bgc_vegetation_inst%InterpFileInputs(bounds_proc)
        call t_stopf('bgc_interp')
+    ! use_fates_spitfire is assigned an integer value in the namelist
+    ! see bld/namelist_files/namelist_definition_clm4_5.xml for details
+    else if (use_fates_spitfire > 1) then
+       call bgc_vegetation_inst%InterpFileInputs(bounds_proc)
```

Comment:
You point to fates_fire_data_method in clmfates_interface

Reply:
In this case I don't understand. Unless I also need to make a copy of
`subroutine InterpFileInputs`
in clmfates_interface, which seems redundant to me.

Reply:
The only thing you actually need from InterpFileInputs is the FATESFireData Interpolation. So there really isn't any duplication of anything in InterpFileInputs. It also HAS to be different than the Vegetation-Facade version -- because the instance of the object is in a different place for FATES than for BGC. The BGC instance isn't actually ever instantiated, so you can't actually use it -- you HAVE to use the version in clmfates_interface.

The reason to create a InterpFileInputs in clmfates_interface is so that the contents of clmfates_interface aren't exposed outside of it. It is a good design goal to keep clmfates_interface content local to itself, so I'd support that method. Technically you could call the FATESFireDdata interpolation call directly here if you exposed the clmfates_interface FATESFireMethod data with a use statement -- but that breaks keeping FATES interface data module to clmfates_interface. So I do recommend having the interface in clmfates_interface.

Reply:
@ekluzek pls let me know if I misunderstood your explanation in the next commit that I'm about to push.